.github/scripts/publish.sh

```diff
@@ -0,0 +1,68 @@
+#!/bin/sh
+set -e
+
+# List of packages to prevent from publishing
+EXCLUDED_PACKAGES=(
+  "storybook"
+)
+
+# List of packages in the workspace
+WORKSPACE_PACKAGES=$(jq -r '.workspaces[]' package.json)
+WORKSPACE_PACKAGES_ARRAY=($WORKSPACE_PACKAGES)
+
+# We substract the excluded packages from the workspace packages
+PACKAGES=()
+for package in "${WORKSPACE_PACKAGES_ARRAY[@]}"; do
+    if [[ ! " ${EXCLUDED_PACKAGES[@]} " =~ " ${package} " ]]; then
+        PACKAGES+=("$package")
+    fi
+done
+
+# Print the filtered list of packages
+echo "::group::Packages to publish"
+echo "${PACKAGES[@]}"
+echo "::endgroup::"
+
+# Get the version from the tag
+VERSION=${GITHUB_REF/refs\/tags\//}
+
+# Helper function to update the version and publish the package
+# Note : we are in the CI triggered by a release, we do not need to create a tag nor commit
+version_and_publish() {
+  local dir=$1
+  local version=$2
+
+  # Go to the package directory
+  pushd $dir
+
+  # Set version in package.json
+  npm version $version --no-git-tag-version
+
+  # Update dependencies to match the new version
+  # This jq script will search for dependencies starting with "@osrd-project" and 
+  # update them to the new version
+  jq \
+    --arg v "$version" \
+    '
+      (.dependencies // {} | with_entries(if .key | startswith("@osrd-project") then .value = $v else . end)) as $d |
+      (.devDependencies // {} | with_entries(if .key | startswith("@osrd-project") then .value = $v else . end)) as $ddev |
+      .dependencies = $d | .devDependencies = $ddev
+    ' \
+    package.json > package.tmp.json
+
+  # Overwrite the package.json with the updated version
+  mv package.tmp.json package.json
```

Comment:
This updates `package.json` but the old version remains in `package-lock.json`, I think? Or does `npm publish` somehow rewrites the lockfile as well?

Reply:
Good point, I'm not sure about this behavior of npm. I'm going to check it out.

Reply:
Seems not to be an issue since the lockfile is handled at the root of the repository and not present in each package. That said, I've forgotten to push the root package lock 😅